src/main/java/use_case/CacheStockInformation/CacheStockInformationInteractor.java

```diff
@@ -30,14 +30,18 @@ public void execute() {
         Set<String> stocksOwned = user.getPortfolio().keySet();
         HashMap<String, List<Double>> stockInformationMap = new HashMap<>();
         for (String stock: stocksOwned) {
-            StockInformation stockInformation = driverAPI.getCurrentStockInformation(stock);
-            List<Double> stockInformationArrayList = new ArrayList<>();
-            stockInformationArrayList.add(stockInformation.getCurrentPrice());
-            stockInformationArrayList.add(stockInformation.getPriceChange());
-            stockInformationArrayList.add(stockInformation.getPercentChange());
-            stockInformationMap.put(
-                    stock, stockInformationArrayList
-            );
+            try {
+                StockInformation stockInformation = driverAPI.getCurrentStockInformation(stock);
```

Comment:
only line 34 needs to be in the try/catch. the others dont have to

Reply:
untrue, since other variables depend on stockInformation, they need to know it exists, which they dont if I only put line 34 in the try catch.

Reply:
<img width="1083" alt="Screenshot 2023-12-01 at 12 33 26 AM" src="https://github.com/awesominat/stonks/assets/75204369/91be02ba-f67b-4c4f-bf84-3d1806a9e535">

Reply:
you forgot to define
`StockInformation stockInformation = null;` at the top

Reply:
That is arguably a worse implementation.
Mine does not require the possibility that StockInformation is null, and simply does nothing if we catch a TickerException. Yours would attempt to call getCurrentPrice on a null type.

Reply:
<img width="1087" alt="Screenshot 2023-12-01 at 1 27 07 AM" src="https://github.com/awesominat/stonks/assets/75204369/99ff5697-064c-49ce-be9c-1870294113aa">

The method still errors out and produces warnings